From b2dd519aadd1fc5f2bc142072029542809eeef6d Mon Sep 17 00:00:00 2001
From: ctang <ctang@cloudera.com>
Date: Fri, 29 Jul 2016 09:15:52 -0400
Subject: [PATCH 0718/1164] HIVE-14359: Hive on Spark might fail in HS2 with
 LDAP authentication in a kerberized cluster
 (Chaoyu Tang, reviewed by Xuefu Zhang)

Change-Id: I49cddb166bb655f7cfabcea1a612a357fbd0234f
---
 .../apache/hive/spark/client/SparkClientImpl.java  |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/spark-client/src/main/java/org/apache/hive/spark/client/SparkClientImpl.java b/spark-client/src/main/java/org/apache/hive/spark/client/SparkClientImpl.java
index dfe263f..59a31c2 100644
--- a/spark-client/src/main/java/org/apache/hive/spark/client/SparkClientImpl.java
+++ b/spark-client/src/main/java/org/apache/hive/spark/client/SparkClientImpl.java
@@ -17,6 +17,8 @@
 
 package org.apache.hive.spark.client;
 
+import static org.apache.hadoop.fs.CommonConfigurationKeysPublic.HADOOP_SECURITY_AUTHENTICATION;
+
 import com.google.common.base.Charsets;
 import com.google.common.base.Joiner;
 import com.google.common.base.Strings;
@@ -326,7 +328,7 @@ public void run() {
 
       List<String> argv = Lists.newArrayList();
 
-      if (hiveConf.getVar(HiveConf.ConfVars.HIVE_SERVER2_AUTHENTICATION).equalsIgnoreCase("kerberos")) {
+      if ("kerberos".equalsIgnoreCase(hiveConf.get(HADOOP_SECURITY_AUTHENTICATION))) {
           argv.add("kinit");
           String principal = SecurityUtil.getServerPrincipal(hiveConf.getVar(ConfVars.HIVE_SERVER2_KERBEROS_PRINCIPAL),
               "0.0.0.0");
-- 
1.7.9.5

